src/main/java/org/apache/sling/auth/oauth_client/impl/OidcAuthenticationHandler.java

```diff
@@ -0,0 +1,509 @@
+        @AttributeDefinition(name = "Path",
+                description = "Repository path for which this authentication handler should be used by Sling. If this is " +
+                        "empty, the authentication handler will be disabled. By default this is set to \"/\".")
+        String path() default "/";
+
+        @AttributeDefinition(name = "Sync Handler Configuration Name",
+                description = "Name of Sync Handler Configuration")
+        String idp() default "oidc";
+
+        @AttributeDefinition(name = "Callback URI",
+                description = "Callback URI")
+        String callbackUri() default "callbackUri";
+
+        @AttributeDefinition(name = "Default Redirect",
+                description = "Default Redirect")
+        String defaultRedirect() default "/";
+
+        @AttributeDefinition(name = "Default Connection Name",
+                description = "Default Connection Name")
+        String defaultConnectionName() default "";
+
+        @AttributeDefinition(name = "UserInfo Enabled",
+                description = "UserInfo Enabled")
+        boolean userInfoEnabled() default true;
+
+    }
+
+    @Activate
+    public OidcAuthenticationHandler(@Reference(policyOption = ReferencePolicyOption.GREEDY) @NotNull SlingRepository repository,
+                                     @NotNull BundleContext bundleContext, @Reference List<ClientConnection> connections,
+                                     @Reference OAuthStateManager stateManager,
+                                     Config config,
+                                     @Reference(cardinality = ReferenceCardinality.OPTIONAL, policyOption = ReferencePolicyOption.GREEDY) LoginCookieManager loginCookieManager,
+                                     @Reference(policyOption = ReferencePolicyOption.GREEDY) UserInfoProcessor userInfoProcessor
+    ) {
+
+        this.repository = repository;
+        this.connections = connections.stream()
+                .collect(Collectors.toMap( ClientConnection::name, Function.identity()));
+        this.stateManager = stateManager;
+        this.idp = config.idp();
+        this.callbackUri = config.callbackUri();
+        this.defaultRedirect = config.defaultRedirect();
+        this.loginCookieManager = loginCookieManager;
+        this.defaultConnectionName = config.defaultConnectionName();
+        this.userInfoProcessor = userInfoProcessor;
+        this.userInfoEnabled = config.userInfoEnabled();
+
+        logger.debug("activate: registering ExternalIdentityProvider");
+        bundleContext.registerService(
+                new String[]{ExternalIdentityProvider.class.getName(), CredentialsSupport.class.getName()}, new OidcIdentityProvider(idp),
+                null);
+
+        logger.info("OidcAuthenticationHandler successfully activated");
+
+    }
+
+
+
+    @Override
+    public AuthenticationInfo extractCredentials(@Nullable HttpServletRequest request, @Nullable HttpServletResponse response) {
```

Comment:
i would suggest to reduce the complexity of this method in order to make it easier to read and maintain.

Reply:
i gave it a try.... @nscendoni , please check if i didn't introduce any bugs :)